libcxx/test/std/iterators/iterator.primitives/range.iter.ops/range.iter.ops.advance/iterator_count_sentinel.pass.cpp

```diff
@@ -42,6 +42,18 @@ constexpr void check_forward(int* first, int* last, std::iter_difference_t<It> n
     // regardless of the iterator category.
     assert(it.stride_count() == M);
     assert(it.stride_displacement() == M);
+    if (n == 0) {
```

Comment:
I would still prefer to pass the expected number of comparisons as an argument to the function -- is it undesirable for some reason? Comments definitely made it clearer what's going on, but my first reaction is still that there's too much logic in this test case.

Reply:
I think I finally understand. I now pass in the expected `stride_count`/`stride_displacement`/`equals_count` as appropriate so that the logic of the tests is now simpler. The tradeoff is at the call site of `check_forward`/`check_backward` where I need to calculate the expected values.

Did you have something like this in mind?

Reply:
I like this better -- IMO placing this logic where the test case is being set up is more natural, and also this is consistent with how `expected` is calculated. It could probably be refactored further but I think we'll be hitting a point of diminishing returns.

Reply:
Agreed, I like it better as well :)